### PR TITLE
Add keypair docs

### LINKED ISSUE #1411: Add more examples to documentation
Add more examples to the documentation to also include more advanced use cases.

### PR DESCRIPTION
**Pull request by kkawula**
Closes #1411

## Introduced changes

- Added docs for `KeyPair` class

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: docs/guide/generating_key_pair.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Key pair
+--------
+
+The Key pair is a pair of private and public keys. The private key is used to sign transactions, and the public key is used to verify the signature.
+In the starknet.py you need to use the :meth:`~starknet_py.net.signer.key_pair.KeyPair` class to be able to create an :meth:`~starknet_py.net.account.account.Account` and :meth:`~starknet_py.net.signer.stark_curve_signer.StarkCurveSigner` object.
+
+Generating random key pair
+--------------------------
+
+Method :meth:`~starknet_py.net.signer.key_pair.KeyPair.generate` generate cryptographically strong pseudo-random numbers
```
Comment (kkawula) [inline docs/guide/generating_key_pair.rst:13]: This method isn't linked in the generated docs. I don't know why. Wdyt? @franciszekjob

Comment (franciszekjob) [inline docs/guide/generating_key_pair.rst:13]: nit:
```suggestion
Method :meth:`~starknet_py.net.signer.key_pair.KeyPair.generate` allows to generate cryptographically strong pseudo-random numbers
```

File: docs/guide/generating_key_pair.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+suitable for managing secrets such as account authentication, tokens, and similar.
+
+.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_key_pair.py
+    :language: python
+    :dedent: 4
+    :start-after: docs-generate: start
+    :end-before: docs-generate: end
+
+
+Creating Key pair from Private Key
```
Comment (franciszekjob) [inline docs/guide/generating_key_pair.rst:23]: nit:
```suggestion
Creating key pair from private Key
```

File: docs/guide/generating_key_pair.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+To create a key pair from a private key, use the :meth:`~starknet_py.net.signer.key_pair.KeyPair.from_private_key` method.
+
+
+.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_key_pair.py
+    :language: python
+    :dedent: 4
+    :start-after: docs-from-private-key: start
+    :end-before: docs-from-private-key: end
+
+Reading Key pair from Keystore file
```
Comment (franciszekjob) [inline docs/guide/generating_key_pair.rst:35]: nit:
```suggestion
Reading key pair from keystore file
```

File: docs/guide/generating_key_pair.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
+Generating a Key pair
+=====================
+
+Key pair
+--------
+
+The Key pair is a pair of private and public keys. The private key is used to sign transactions, and the public key is used to verify the signature.
+In the starknet.py you need to use the :meth:`~starknet_py.net.signer.key_pair.KeyPair` class to be able to create an :meth:`~starknet_py.net.account.account.Account` and :meth:`~starknet_py.net.signer.stark_curve_signer.StarkCurveSigner` object.
```
Comment (franciszekjob) [inline docs/guide/generating_key_pair.rst:8]:
```suggestion
In the starknet.py you need to use the :class:`~starknet_py.net.signer.key_pair.KeyPair` class to be able to create an :class:`~starknet_py.net.account.account.Account` and :class:`~starknet_py.net.signer.stark_curve_signer.StarkCurveSigner` object.
```

File: docs/guide/generating_key_pair.rst (hunk 1 above)
Comment (franciszekjob) [reply docs/guide/generating_key_pair.rst]: offline discussion: Looks like using ``:meth:`~starknet_py.net.signer.stark_curve_signer.KeyPair.generate`` resolves the issue, this is very strange though.

**Review by franciszekjob — APPROVED** [review]
Chuck Norris Thumbs Up
LGTM

**Pull request merged**